### PR TITLE
selectAccountDataByType return ClientEvent pointer instead of slice of ClientEvent

### LINKED ISSUE #773: selectAccountDataByType shouldn't return a slice
[`selectAccountDataByType`](https://github.com/matrix-org/dendrite/blob/3578d77d259c852a16dc430725d348e2c62c4ff5/clientapi/auth/storage/accounts/account_data_table.go#L121) currently returns a `[]gomatrixserverlib.ClientEvent`. However since there can only be [one](https://github.com/matrix-org/dendrite/blob/3578d77d259c852a16dc430725d348e2c62c4ff5/clientapi/auth/storage/accounts/account_data_table.go#L42) account data row for a given `(localpart, room_id, type)` tuple, we can either change the return type to a pointer or a plain `ClientEvent` with an error to indicate empty data. This should also make the return value less confusing.

### PR DESCRIPTION
**Pull request by behouba**
### Pull Request Checklist

This pull request is an attempt to fix #773. 

Signed-off-by: Kouame Behouba Manassé behouba@gmail.com
* [ ] I have added any new tests that need to pass to `testfile` as specified in [docs/sytest.md](https://github.com/matrix-org/dendrite/blob/master/docs/sytest.md)
* [x] Pull request includes a [sign off](https://github.com/matrix-org/dendrite/blob/master/CONTRIBUTING.md#sign-off)

### CODE REVIEW

File: clientapi/auth/storage/accounts/account_data_table.go [hunk 1]
```diff
@@ -120,28 +120,20 @@ func (s *accountDataStatements) selectAccountData(
 
 func (s *accountDataStatements) selectAccountDataByType(
 	ctx context.Context, localpart, roomID, dataType string,
-) (data []gomatrixserverlib.ClientEvent, err error) {
-	data = []gomatrixserverlib.ClientEvent{}
-
+) (data *gomatrixserverlib.ClientEvent, err error) {
 	stmt := s.selectAccountDataByTypeStmt
-	rows, err := stmt.QueryContext(ctx, localpart, roomID, dataType)
+	row := stmt.QueryRowContext(ctx, localpart, roomID, dataType)
```
Comment (Cnly) [inline clientapi/auth/storage/accounts/account_data_table.go]:
```suggestion
```

This line and the `Scan` statement below can be merged into the `if` block.

File: clientapi/auth/storage/accounts/account_data_table.go [hunk 2]
```diff
@@ -120,28 +120,20 @@ func (s *accountDataStatements) selectAccountData(
...
+
```
Comment (Cnly) [inline clientapi/auth/storage/accounts/account_data_table.go]:
```suggestion
```

File: clientapi/auth/storage/accounts/account_data_table.go [hunk 3]
```diff
@@ -120,28 +120,20 @@ func (s *accountDataStatements) selectAccountData(
...
+	var content []byte
+
+	err = row.Scan(&content)
```
Comment (Cnly) [inline clientapi/auth/storage/accounts/account_data_table.go]:
```suggestion
```

File: clientapi/auth/storage/accounts/account_data_table.go [hunk 4]
```diff
@@ -120,28 +120,20 @@ func (s *accountDataStatements) selectAccountData(
...
 	if err != nil {
```
Comment (Cnly) [inline clientapi/auth/storage/accounts/account_data_table.go]:
```suggestion
	if err = stmt.QueryRowContext(ctx, localpart, roomID, dataType).Scan(&content); err != nil {
```

File: clientapi/auth/storage/accounts/account_data_table.go (hunk 1 above)
Comment (behouba) [reply clientapi/auth/storage/accounts/account_data_table.go]: indeed it would be much better, i will change that.

**Pull request merged**